packages/node/src/indexer/project.service.ts

```diff
@@ -111,6 +124,50 @@ export class ProjectService {
         await this.poiService.init(this.schema);
       }
     }
+
+    await this.resumeUnfinalizedBlocks();
+  }
+
+  private async resumeUnfinalizedBlocks(): Promise<void> {
+    // unfinalized blocks
+    const startUnfinalizedBlocks = await this.getMetadataUnfinalizedBlocks();
+    const lastFinalizedVerifiedHeight =
+      await this.getLastFinalizedVerifiedHeight();
+    if (this.nodeConfig.unfinalizedBlocks) {
+      this.unfinalizedBlockService.init(
+        this.metadataRepo,
+        startUnfinalizedBlocks ?? {},
+        lastFinalizedVerifiedHeight,
+      );
+    } else {
+      // Has previous indexed with unfinalized blocks, but it is no longer enabled
+      if (
```

Comment:
in case `Has previous indexed with unfinalized blocks, but it is no longer enabled`, I can think of two possibilities
We check the finality of these blocks, 
1. if they are no forks. then we remove them from unfinalised block array in metadata and continue index
2. if there are forks, we rewind to the height before the earlest block in them.

In this method, the `lastFinalizedVerifiedHeight` is read from metadata, not from latest network, which could be improved.